code/modules/reagents/reagent_containers/cups/glassbottle.dm

```diff
@@ -473,30 +478,80 @@
 	if(spillable)
 		return ..()
 	balloon_alert(user, "fiddling with cork...")
 	if(do_after(user, 1 SECONDS, src))
-		return pop_cork(user)
+		return pop_cork(user, sabrage = FALSE)
+
+/obj/item/reagent_containers/cup/glass/bottle/champagne/attackby(obj/item/attacking_item, mob/living/user, params)
+	. = ..()
+	if(spillable)
+		return
+	if(attacking_item.get_sharpness())
+		if(attacking_item.force < 5)
+			balloon_alert(user, "not strong enough!")
+			return
+		else
+			playsound(user, 'sound/items/unsheath.ogg', 25, TRUE)
+			balloon_alert(user, "preparing to swing...")
+			if(do_after(user, 2 SECONDS, src)) //takes longer because you are supposed to take the foil off the bottle first
+				//calculate success chance. example: captain's sabre - 15 force = 75% chance
+				if(prob(attacking_item.force * sabrage_success_percentile + \
+				((user.mind.assigned_role.departments_bitflags & (DEPARTMENT_BITFLAG_COMMAND)) ? 20 : 0) + \
+				((HAS_TRAIT(user, TRAIT_SABRAGE_PRO)) ? 35 : 0)
+				))
+					return pop_cork(user, sabrage = TRUE)
+				else //you dun goofed
+					user.visible_message(span_danger("[user] fumbles the sabrage and cuts [src] in half, spilling it over themselves!"), \
+						span_danger("You fail your stunt and cut [src] in half, spilling it over you!"))
+					user.add_mood_event("sabrage_fail", /datum/mood_event/sabrage_fail)
+					return smash(user, user, ranged = FALSE, break_top = TRUE)
 
 /obj/item/reagent_containers/cup/glass/bottle/champagne/update_icon_state()
 	. = ..()
 	if(spillable)
-		icon_state = "[base_icon_state]_popped"
+		if(sabraged)
+			icon_state = "[base_icon_state]_sabrage"
+		else
+			icon_state = "[base_icon_state]_popped"
 	else
 		icon_state = base_icon_state
 
-/obj/item/reagent_containers/cup/glass/bottle/champagne/proc/pop_cork(mob/user)
-	user.visible_message(span_danger("[user] loosens the cork of [src] causing it to pop out of the bottle with great force."), \
-		span_nicegreen("You elegantly loosen the cork of [src] causing it to pop out of the bottle with great force."))
+/obj/item/reagent_containers/cup/glass/bottle/champagne/proc/pop_cork(mob/living/user, sabrage)
+	if(sabrage)
+		sabraged = TRUE
+		user.visible_message(span_danger("[user] cleanly slices off the cork of [src], causing it to fly off the bottle with great force."), \
+			span_nicegreen("You elegantly slice the cork off of [src], causing it to fly off the bottle with great force."))
+		for(var/mob/living/carbon/stunt_witness in view(7, user))
+			for(var/find_previous_showoff in stunt_witness.mob_mood.mood_events) //if someone who did a sabrage before us saw us doing it, clear their success mood event
+				if(find_previous_showoff == "sabrage_success")
+					stunt_witness.clear_mood_event(find_previous_showoff)
+					continue
+			if(stunt_witness == user)
+				stunt_witness.add_mood_event("sabrage_success", /datum/mood_event/sabrage_success)
+				continue
+			else
```

Comment:
if it continues it doesn't execute the following code, you don't need an else case here